frontend/src/App.test.tsx

```diff
@@ -475,3 +480,99 @@ describe("App.handleNewReport", () => {
     expect(SessionInfo.isSet()).toBe(true)
   })
 })
+
+// Using this to test the functionality provided through streamlit.experimental_set_query_params.
+// Please see https://github.com/streamlit/streamlit/issues/2887 for more context on this.
+describe("App.handlePageInfoChanged", () => {
+  beforeEach(() => {
+    // Reset the value of document.location.pathname.
+    window.history.pushState({}, "", "/")
```

Comment:
Hm, I think we'll also want to do this one last time in an `afterAll` just to be certain things get cleaned up in the end (we technically aren't supposed to assume that tests will run in the order they're defined even though that happens in practice)